tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+namespace Keboola\Test\Backend\ExternalBuckets;
+
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Workspaces;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+use Keboola\Test\Backend\Workspaces\Backend\TeradataWorkspaceBackend;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+
+class TeradataRegisterBucketTest extends BaseExternalBuckets
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $token = $this->_client->verifyToken();
+
+        $this->thisBackend = $token['owner']['defaultBackend'];
+
+        if (!in_array('external-buckets', $token['owner']['features'])) {
+            $this->markTestSkipped(sprintf('External buckets are not enabled for project "%s"', $token['owner']['id']));
+        }
+        $this->allowTestForBackendsOnly([self::BACKEND_TERADATA], 'Backend has to support external buckets');
+    }
+
+    public function testInvalidDBToRegister(): void
+    {
+        $this->dropBucketIfExists($this->_client, 'in.test-bucket-registration', true);
+
+        try {
+            $this->_client->registerBucket(
+                'test-bucket-registration',
+                ['non-existing-database'],
+                'in',
+                'will fail',
+                $this->thisBackend,
+                'test-bucket-will-fail'
+            );
+        } catch (ClientException $e) {
+            $this->assertSame('storage.dbObjectNotFound', $e->getStringCode());
+            $this->assertStringContainsString(
+                'doesn\'t exist or project user is missing privileges to read from it.',
+                $e->getMessage()
+            );
+        }
+    }
+
+    public function testRegisterWSAsExternalBucket(): void
+    {
+        $ws = new Workspaces($this->_client);
+
+        $workspace = $ws->createWorkspace();
+
+        $runId = $this->setRunId();
+        $idOfBucket = $this->_client->registerBucket(
+            'test-bucket-registration',
+            [$workspace['connection']['schema']],
+            'in',
+            'Iam in workspace',
+            $this->thisBackend,
+            'Iam-your-workspace'
+        );
+        $this->assertEvents($runId, ['storage.bucketCreated']);
```

Comment:
uz mozes pouzivat nove assertovanie event 
```
$this->initEvents($this->_client);
 $assertCallback = function ($events) {
            $this->assertCount(1, $events);
        };
        $query = new EventsQueryBuilder();
        $query->setEvent('storage.bucketCreated')
            ->setRunId($runId);
        $this->assertEventWithRetries($this->_client, $assertCallback, $query);
```
https://github.com/keboola/storage-api-php-client/pull/979/files

Reply:
je to tam :)

Reply:
A tu na riadku 69 to nejde pouzit ?